tests/virtctl/ssh.go

```diff
@@ -0,0 +1,94 @@
+package virtctl
+
+import (
+	"crypto/ecdsa"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"golang.org/x/crypto/ssh"
+
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/clientcmd"
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/flags"
+	"kubevirt.io/kubevirt/tests/libvmi"
+	"kubevirt.io/kubevirt/tests/util"
+)
+
+var _ = Describe("[sig-compute][virtctl]SSH", func() {
+	var pub ssh.PublicKey
+	var keyFile string
+	var virtClient kubecli.KubevirtClient
+
+	BeforeEach(func() {
+		var err error
+		virtClient, err = kubecli.GetKubevirtClient()
+		Expect(err).ToNot(HaveOccurred())
+		// Disable SSH_AGENT to not influence test results
+		Expect(os.Setenv("SSH_AUTH_SOCK", "/dev/null")).To(Succeed())
+		keyFile = filepath.Join(GinkgoT().TempDir(), "id_rsa")
+		tests.BeforeTestCleanup()
+		var priv *ecdsa.PrivateKey
+		priv, pub, err = NewKeyPair()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(DumpPrivateKey(priv, keyFile)).To(Succeed())
+	})
+
+	It("should succeed to execute a command on the VM using the ssh native method", func() {
+		By("injecting a SSH public key into a VMI")
+		vmi := libvmi.NewCirros(
+			libvmi.WithCloudInitNoCloudUserData(renderUserDataWithKey(pub), false))
+		vmi, err := virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(vmi)
+		Expect(err).ToNot(HaveOccurred())
+
+		vmi = tests.WaitUntilVMIReady(vmi, console.LoginToCirros)
+		By("ssh into the VM using the native option")
+		Expect(clientcmd.NewRepeatableVirtctlCommand(
+			"ssh",
+			"--local-ssh=false",
+			"--namespace", util.NamespaceTestDefault,
+			"--username", "cirros",
+			"--identity-file", keyFile,
+			"--known-hosts=",
+			`--command='true'`,
+			vmi.Name)()).To(Succeed())
+
+	})
+	It("should succeed execute a command on the VM using local ssh method", func() {
+		By("injecting a SSH public key into a VMI")
+		vmi := libvmi.NewCirros(
+			libvmi.WithCloudInitNoCloudUserData(renderUserDataWithKey(pub), false))
+		vmi, err := virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(vmi)
+		Expect(err).ToNot(HaveOccurred())
+
+		vmi = tests.WaitUntilVMIReady(vmi, console.LoginToCirros)
+		By("ssh into the VM using the native option")
+		cmd := exec.Command("virtctl",
+			"ssh",
+			"--local-ssh=true",
+			"--namespace", util.NamespaceTestDefault,
+			"--username", "cirros",
+			"--identity-file", keyFile,
+			"-t", "-o StrictHostKeyChecking=no",
+			"-t", "-o UserKnownHostsFile=/dev/null",
+			"--command", "true",
+			vmi.Name)
+		envPath := os.Getenv("PATH")
+		if envPath != "" {
+			envPath = fmt.Sprintf("PATH=%s:%s", envPath, flags.KubeVirtVirtctlPath)
```

Comment:
After greping the sources it seems to me that `flags.KubeVirtVirtctlPath` is not actually used anywhere and by default is emply. So probably there is a need to pass it to the testsuite... And TBH its not fully clear to me what the semanthics should be: whether the var should look like `/path/to/virtctl` or just `/path/to`. 

Also perhaps its worth using `SkipIfNoCmd("virtctl")` in `BeforeEach`.

Reply:
> After greping the sources it seems to me that `flags.KubeVirtVirtctlPath` is not actually used anywhere and by default is emply. So probably there is a need to pass it to the testsuite... And TBH its not fully clear to me what the semanthics should be: whether the var should look like `/path/to/virtctl` or just `/path/to`.
> 

I expected to be `/path/to` but I simply tested locally and I have virtctl installed, I just wanted to verify how it was behaving in the upstream CI

> Also perhaps its worth using `SkipIfNoCmd("virtctl")` in `BeforeEach`.

I'd like to test it because if by chance virtctl is never installed then we don't run this test

Reply:
mmh in kubevirtci it seems that this flag is never set :/

Reply:
Yeah, I also had that impression after looking at the sources. But I think this can be set in `hack/functests.sh`. The tesstuite itself is run as `_out/tests/ginkgo ...`. I would assume the `virtctl` binary can then be found at `_out/cmd/virtctl/virtctl`.